sklearn/utils/graph.py

```diff
@@ -119,18 +122,26 @@ def graph_laplacian(csgraph, normed=False, return_diag=False):
 
     if sparse.isspmatrix(csgraph):
         return _laplacian_sparse(csgraph, normed=normed,
-                                 return_diag=return_diag)
+                                 return_diag=return_diag, copy=copy)
     else:
         return _laplacian_dense(csgraph, normed=normed,
-                                return_diag=return_diag)
+                                return_diag=return_diag, copy=copy)
 
 
-def _laplacian_sparse(graph, normed=False, return_diag=False):
+def _laplacian_sparse(graph, normed=False, return_diag=False, copy=True):
     n_nodes = graph.shape[0]
-    if not graph.format == 'coo':
-        lap = (-graph).tocoo()
+    if not copy:  # use the same matrix
+        np.negative(graph, out=graph)
```

Comment:
At this point, graph is a sparse matrix, and I doubt this will work in all supported versions of scipy. You should convert to `coo` first, and then  perform the inplace negation on its `data`.

Although, really, sparse matrices are probably not the problem in terms of memory errors, so I would be okay with leaving the sparse case unchanged.